env0/data_credentials.go

```diff
@@ -0,0 +1,63 @@
+package env0
+
+import (
+	"context"
+	"strings"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataCredentials(cloudType string) *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataCredentialsRead(cloudType),
```

Comment:
notice that the read context receives a cloudType variable.
This could be aws, azure or gcp (in the future could be expanded to additional types).